src/persist-client/src/operators/shard_source.rs

```diff
@@ -610,5 +597,9 @@ where
         }
     });
 
-    (fetched_stream, completed_fetches_stream)
+    (
+        fetched_stream,
+        completed_fetches_stream,
+        Rc::new(shutdown_button),
```

Comment:
This needs to be `shutdown_button.press_on_drop()` to make it trigger on drop

Reply:
yuuuuuuuuup good catch